cmd/trust-manager/app/ciphers_test.go

```diff
@@ -0,0 +1,259 @@
+/*
+Copyright 2025 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"log"
+	"math/big"
+	"net"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/spf13/cobra"
+	cliflag "k8s.io/component-base/cli/flag"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	"github.com/cert-manager/trust-manager/cmd/trust-manager/app/options"
+)
+
+func TestCiphersSuite(t *testing.T) {
+	tmpDir := t.TempDir()
+	err := setupWebHookServer(tmpDir)
+	if err != nil {
+		t.Fatal(fmt.Errorf("unable to setup webhook server %s\n", err))
+	}
+
+	t.Run("TLS 1.2 client connect to TLS 1.2 server, ciphers suite supported", func(t *testing.T) {
+		client := http.Client{
+			Transport: &http.Transport{
+				ForceAttemptHTTP2: false,
+				TLSClientConfig: &tls.Config{
+					InsecureSkipVerify: true, //nolint: gosec
+					MinVersion:         tls.VersionTLS12,
+					MaxVersion:         tls.VersionTLS12,
+					CipherSuites: []uint16{
+						tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+						tls.TLS_RSA_WITH_AES_128_GCM_SHA256,
+					},
+				},
+			},
+		}
+
+		resp, err := client.Get(fmt.Sprintf("https://%s", net.JoinHostPort("localhost", "6443"))) //nolint: noctx
+		if err != nil {
+			t.Fatalf("error to connect to webhook server, %s\n", err)
+		}
+		resp.Body.Close()
```

Comment:
suggestion: If we call `Fatalf` the Body will never be closed - as far as I know at least!

```suggestion
		resp, err := client.Get(fmt.Sprintf("https://%s", net.JoinHostPort("localhost", "6443"))) //nolint: noctx
		defer resp.Body.Close()
		if err != nil {
			t.Fatalf("error to connect to webhook server, %s\n", err)
		}
```

Reply:
Yes, you are right. That was my omission.